dist/MenuItem.d.ts

```diff
@@ -19,9 +21,12 @@ declare global {
             'sp-menu-item': {
                 children?: React.ReactNode;
                 ref?: React.RefObject<HTMLElement>;
+                key?: string;
```

Comment:
Good call on adding key to the list types!

Reply:
Yes, a few months ago I had to add "key" to the menu item component in order to update its state correctly. But we can't simply have "key" property on the component since it will be undefined. We have to find a better way.

Reply:
The beauty of react here is that `key={undefined}` will actually just not render a key at all!
I use the same trick for making sure false doesn't add an attribute to the webcomponent when the spec calls for not rendering the key.

Alternatively, we could also just generate keys if it is not passed in, so that we don't get react warnings about missing keys in lists...

Maybe I should whip up some tests around these missing keys... 🤔